### PR TITLE
support partial clones

### PR DESCRIPTION
**Pull request by chm-diederichs**

### CODE REVIEW

File: lib/core.js [hunk 1]
```diff
@@ -227,7 +227,7 @@ module.exports = class Core {
         const segmentStart = pos
         const segmentEnd = src.bitfield.firstUnset(pos)
 
-        if (segmentStart < 0) break
+        if (segmentStart > length || segmentStart < 0) break
```
Comment (mafintosh) [inline lib/core.js:230]: >= ?

File: lib/core.js [hunk 2]
```diff
@@ -217,7 +217,7 @@ module.exports = class Core {
     return false
   }
 
-  async copyFrom (src, signature) {
+  async copyFrom (src, signature, length = src.tree.length) {
```
Comment (mafintosh) [inline lib/core.js:220]: make the length an option, { length = src.tree.length } = {}

**Comment by chm-diederichs** [comment]
hold off on merging this, this fixes an autobase bug, which should probably be fixed in autobase

File: lib/core.js [hunk 3]
```diff
@@ -247,7 +247,7 @@ module.exports = class Core {
       // TODO: is it an issue to move the nodes directly?
       // TODO: make flat iterator that computes needed nodes
 
-      for (let i = 0; i <= src.tree.length * 2; i++) {
+      for (let i = 0; i <= length * 2; i++) {
```
Comment (mafintosh) [inline lib/core.js:250]: should be `<` not `<=` i think, unrelated

**Pull request merged**